internal/cmd/gocdk/resource.go

```diff
@@ -113,6 +113,47 @@ var provisionableTypes = map[string]func(*processContext, string) ([]*static.Act
 		}, nil
 	},
 
+	// DOCSTORE
+	"docstore/awsdynamodb": func(pctx *processContext, biomeDir string) ([]*static.Action, error) {
+		reader := bufio.NewReader(pctx.stdin)
+		_, err := prompt.AWSRegionIfNeeded(reader, pctx.stderr, biomeDir)
+		if err != nil {
+			return nil, err
+		}
+		return []*static.Action{
+			static.AddProvider("aws"),
+			static.AddOutputVar("DOCSTORE_COLLECTION_URL", "${local.awsdynamodb_url}"),
+			static.CopyFile("/resource/docstore/awsdynamodb.tf", "awsdynamodb.tf"),
+		}, nil
+	},
+	"docstore/azurecosmos": func(pctx *processContext, biomeDir string) ([]*static.Action, error) {
+		reader := bufio.NewReader(pctx.stdin)
+		_, err := prompt.AzureLocationIfNeeded(reader, pctx.stderr, biomeDir)
+		if err != nil {
+			return nil, err
+		}
+		return []*static.Action{
+			static.AddProvider("azurerm"),
+			static.AddProvider("random"),
```

Comment:
What does "random" mean here?

Reply:
It means it will add the `random` provider (if it hasn't already been added yet); it is used to generate a random name for various things (e.g., the AWS table, the Azure Cosmos account, the Firestore database name).

```
  name         = local.gocdk_random_name
```

is a reference to that provider.